src/sidebar/services/streamer.js

```diff
@@ -21,6 +22,23 @@ export default function Streamer(store, auth, groups, session, settings) {
   // established.
   const configMessages = {};
 
+  let reconnectSetUp = false;
+
+  /**
+   * Set up automatic reconnecting when user changes.
+   */
+  function setUpAutoReconnect() {
+    if (reconnectSetUp) {
+      return;
+    }
+    reconnectSetUp = true;
+
+    // Reconnect when user changes, as auth token will have changed
+    watch(store.subscribe, [() => store.profile().userid], () => {
```

Comment:
This doesn't seem to suffer from the first-profile-load problem, but I could add that check here once #2101 lands if it seems needed.

Reply:
Since the connection is initiated by `SidebarContent`, which is only rendered after the profile is initially fetched, this doesn't suffer from the issue the groups service had where it would trigger unnecessarily on startup when an existing login session was resumed.